devito/mpi/distributed.py

```diff
@@ -583,15 +584,30 @@ class CustomTopology(tuple):
 
     * `(1, '*', 1)`: the wildcard `'*'` tells the runtime to decompose the y
                      Dimension into N chunks
-    * `('*', '*', 1)`: the wildcard `'*'` tells the runtime to decompose both the
-                       x and y Dimensions into N / 2 chunks respectively.
-
-    Raises
-    ------
-    N must evenly divide the number of `'*'`, otherwise a ValueError exception
-    is raised.
-    If the wildcard `'*'` is used, then the CustomTopology can only contain either
-    `'*'` or 1's, otherwise a ValueError exception is raised.
+    * `('*', '*', 1)`: the wildcard `'*'` tells the runtime to decompose both
+                       the x and y Dimensions in `nstars` factors of N, prioritizing
+                       the outermost dimension
+
+    Assuming N=6 and requested topology is `('*', '*', 1)`,
+    since there is no integer k, so that k*k=6, we resort to the closest factors to
```

Comment:
Could you rewrite this (grammarly maybe)? it's not very clear to me at first glance